purchases/src/customEntitlementComputation/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -0,0 +1,365 @@
+         * By default, LogLevel.DEBUG in debug builds, and LogLevel.INFO in release builds.
+         */
+        @JvmStatic
+        var logLevel: LogLevel
+            get() = PurchasesOrchestrator.logLevel
+            set(value) {
+                PurchasesOrchestrator.logLevel = value
+            }
+
+        /**
+         * Set a custom log handler for redirecting logs to your own logging system.
+         * Defaults to [android.util.Log].
+         *
+         * By default, this sends info, warning, and error messages.
+         * If you wish to receive Debug level messages, see [debugLogsEnabled].
+         */
+        @JvmStatic
+        var logHandler: LogHandler
+            @Synchronized get() = PurchasesOrchestrator.logHandler
+
+            @Synchronized set(value) {
+                PurchasesOrchestrator.logHandler = value
+            }
+
+        @JvmSynthetic
+        internal var backingFieldSharedInstance: Purchases? = null
+
+        /**
+         * Singleton instance of Purchases. [configure] will set this
+         * @return A previously set singleton Purchases instance
+         * @throws UninitializedPropertyAccessException if the shared instance has not been configured.
+         */
+        @JvmStatic
+        var sharedInstance: Purchases
+            get() =
+                backingFieldSharedInstance
+                    ?: throw UninitializedPropertyAccessException(ConfigureStrings.NO_SINGLETON_INSTANCE)
+
+            @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+            internal set(value) {
+                backingFieldSharedInstance?.close()
+                backingFieldSharedInstance = value
+            }
+
+        /**
+         * Current version of the Purchases SDK
+         */
+        @JvmStatic
+        val frameworkVersion = PurchasesOrchestrator.frameworkVersion
+
+        /**
+         * Set this property to your proxy URL before configuring Purchases *only*
+         * if you've received a proxy key value from your RevenueCat contact.
+         */
+        @JvmStatic
+        var proxyURL: URL?
+            get() = PurchasesOrchestrator.proxyURL
+            set(value) { PurchasesOrchestrator.proxyURL = value }
+
+        /**
+         * True if [configure] has been called and [Purchases.sharedInstance] is set
+         */
+        @JvmStatic
+        val isConfigured: Boolean
+            get() = this.backingFieldSharedInstance != null
+
+        /**
+         * Configures an instance of the Purchases SDK with a specified API key. The instance will
+         * be set as a singleton. You should access the singleton instance using [Purchases.sharedInstance]
+         * @param configuration: the [PurchasesConfiguration] object you wish to use to configure [Purchases].
+         * @return An instantiated `[Purchases] object that has been set as a singleton.
+         */
+        @JvmStatic
+        fun configure(
+            configuration: PurchasesConfiguration,
+        ): Purchases {
```

Comment:
you could still configure this without an appUserID, right? we should make it an explicit parameter

Reply:
disregard that comment, I just saw the other PR where it becomes a mandatory param of PurchasesConfiguration